packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -193,6 +193,33 @@ abstract class Domain {
 
   void _send(Map<String, dynamic> map) => daemon._send(map);
 
+  String _getArgAsString(Map<String, dynamic> args, String name, { bool required: false }) {
+    if (required && !args.containsKey(name))
+      throw "$name is required";
+    dynamic val = args[name];
+    if (val != null && val is! String)
+      throw "$name is not a string";
+    return val;
+  }
+
+  bool _getArgAsBool(Map<String, dynamic> args, String name, { bool required: false }) {
```

Comment:
The "As" in the function name makes me think the value is parsed into a `bool`. How about `_getBoolArg`?